app/src/main/java/com/zeapo/pwdstore/PasswordStore.kt

```diff
@@ -554,41 +557,91 @@ class PasswordStore : AppCompatActivity(R.layout.activity_pwdstore) {
+            .setMessage(resources.getQuantityString(R.plurals.delete_dialog_text, size, size))
             .setPositiveButton(resources.getString(R.string.dialog_yes)) { _, _ ->
-                val filesToDelete = if (item.file.isDirectory) {
-                    item.file.listFilesRecursively()
-                } else {
-                    listOf(item.file)
+                val filesToDelete = arrayListOf<File>()
+                selectedItems.forEach { item ->
+                    if (item.file.isDirectory)
+                        filesToDelete.addAll(item.file.listFilesRecursively())
+                    else
+                        filesToDelete.add(item.file)
                 }
+                selectedItems.map { item -> item.file.deleteRecursively() }
                 AutofillMatcher.updateMatches(applicationContext, delete = filesToDelete)
-                item.file.deleteRecursively()
-                commitChange(resources.getString(R.string.git_commit_remove_text, item.longName))
-                deletePasswords(selectedItems)
-            }
-            .setNegativeButton(resources.getString(R.string.dialog_no)) { _, _ ->
-                deletePasswords(selectedItems)
+                commitChange(resources.getString(R.string.git_commit_remove_text,
+                    selectedItems.joinToString(separator = ", ") { item ->
+                        item.file.toRelativeString(getRepositoryDirectory(this))
+                    }
+                ))
+                refreshPasswordList()
             }
+            .setNegativeButton(resources.getString(R.string.dialog_no), null)
             .show()
     }
 
     fun movePasswords(values: List<PasswordItem>) {
         val intent = Intent(this, SelectFolderActivity::class.java)
-        val fileLocations = ArrayList<String>()
-        for ((_, _, _, file) in values) {
-            fileLocations.add(file.absolutePath)
-        }
+        val fileLocations = values.map { it.file.absolutePath }.toTypedArray()
         intent.putExtra("Files", fileLocations)
         intent.putExtra(BaseGitActivity.REQUEST_ARG_OP, "SELECTFOLDER")
-        startActivityForResult(intent, REQUEST_CODE_SELECT_FOLDER)
+        registerForActivityResult(StartActivityForResult()) { result ->
+            val intentData = result.data ?: return@registerForActivityResult
+            d { "Moving passwords to ${intentData.getStringExtra("SELECTED_FOLDER_PATH")}" }
+            d { TextUtils.join(", ", requireNotNull(intentData.getStringArrayExtra("Files"))) }
+
+            val target = File(requireNotNull(intentData.getStringExtra("SELECTED_FOLDER_PATH")))
+            val repositoryPath = getRepositoryDirectory(applicationContext).absolutePath
+            if (!target.isDirectory) {
+                e { "Tried moving passwords to a non-existing folder." }
+                return@registerForActivityResult
+            }
+
+            lifecycleScope.launch(Dispatchers.IO) {
```

Comment:
I know it's my favourite kind of concern, but what happens if the activity is destroyed in the middle of a move operation? Should we at least show some UI to let the user know something is being done in the background?

Reply:
Not particularly sure how to best convey it tbh. I don't want to bring the progress dialog back :/

Reply:
I have an idea, working on a commit.

Reply:
Nvm, the file operations are way too fast for this to matter.